lib/Target/LLVMIR/LLVMDIScope.cpp

```diff
@@ -74,16 +74,24 @@ struct LLVMDIScopePass : public LLVMDIScopeBase<LLVMDIScopePass> {
           context, llvm::sys::path::filename(inputFilePath),
           llvm::sys::path::parent_path(inputFilePath));
     }
-    auto distinctId = mlir::DistinctAttr::create(mlir::UnitAttr::get(context));
-    if (!compileUnitAttr) {
-      compileUnitAttr = LLVM::DICompileUnitAttr::get(
-          context, distinctId, llvm::dwarf::DW_LANG_C, fileAttr,
-          StringAttr::get(context, "triton"),
-          /*isOptimized=*/true, LLVM::DIEmissionKind::LineTablesOnly);
-    }
     auto subroutineTypeAttr =
         LLVM::DISubroutineTypeAttr::get(context, llvm::dwarf::DW_CC_normal, {});
 
+    DistinctAttr distinctId;
```

Comment:
This block is not obvious to me.  Could we add a comment explaining why we do what we do here?

Reply:
Thank you for the explanation.

I think part of what's confusing in this block is that it's not obvious to me what are the in/out variables here.  A short comment introducing the block of code would help me.

Reply:
Done.